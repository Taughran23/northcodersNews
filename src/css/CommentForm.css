.comment-form {
    background-color:white;
}

.form {
    padding:15px
}

.button {
    margin-top: 15px;
    color:white;
    background-color: black
}