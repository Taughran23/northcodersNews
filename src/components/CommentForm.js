import React from 'react';
import PropTypes from 'prop-types';

import '../css/CommentForm.css';

const CommentForm = (props) => (
  <div className='comment-form'>
    <form className='form' onSubmit={props.submitHandler}>
      <textarea
        className="textarea"
        placeholder="Post comment"
        value={props.input}
        onChange={props.inputHandler}
      />
      <button
        className="button"
        type="submit"
        value="post"
      >
      Post comment
      </button>
    </form>

  </div>   
);

CommentForm.propTypes = {
  input: PropTypes.string.isRequired,
  submitHandler: PropTypes.func.isRequired, 
  inputHandler: PropTypes.func.isRequired
};

export default CommentForm;