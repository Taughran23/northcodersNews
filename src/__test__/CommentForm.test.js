import React from 'react';
import Enzyme , {shallow} from 'enzyme';
import CommentForm  from '../components/CommentForm';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';

Enzyme.configure({ adapter: new Adapter() });

describe('<CommentForm />', () => {
    
  it('should render correctly', () => {
    const component = shallow(<CommentForm
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
  it('it renders 1 <CommentForm /> component', () => {
    const component = shallow(<CommentForm />);
    expect(component).toHaveLength(1);
  });
});