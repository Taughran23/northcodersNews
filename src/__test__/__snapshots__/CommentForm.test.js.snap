// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> should render correctly 1`] = `
<div
  className="comment-form"
>
  <form
    className="form"
  >
    <textarea
      className="textarea"
      placeholder="Post comment"
    />
    <button
      className="button"
      id="post-button"
      type="submit"
      value="post"
    >
      Post comment
    </button>
  </form>
</div>
`;
